### ISSUE TITLE
MEMPTR memory leak - false positives

### ISSUE THREAD

**Issue by Ton-Pe** [Riverside-Software/sonar-openedge#1125]
### Case 1:
Following code gives a false positive. Size of memptr is set inside the block and then released outside that block.
```
DEFINE VARIABLE mReceiveData AS MEMPTR   NO-UNDO.

IF TRUE THEN DO:
    SET-SIZE(mReceiveData) = 100.
END.

SET-SIZE(mReceiveData) = 0.
```

If release is done inside the same block or moved to finally-block, then issue is not raised.

### Case 2:
Following code where copy-lob is done inside a nested block, gives false positive.
```
DEFINE VARIABLE mReceiveData AS MEMPTR   NO-UNDO.
DEFINE VARIABLE mSendData    AS MEMPTR   NO-UNDO.

IF TRUE THEN DO: // First level
    DO: // second level
        COPY-LOB mReceiveData TO mSendData.
    END.
END.


FINALLY:
    SET-SIZE (mSendData) = 0.
END FINALLY.
```

For a comparison, this code, where there is only one block, issue is not raised:

```
DEFINE VARIABLE mReceiveData AS MEMPTR   NO-UNDO.
DEFINE VARIABLE mSendData    AS MEMPTR   NO-UNDO.

IF TRUE THEN DO:
        COPY-LOB mReceiveData TO mSendData.
END.

FINALLY:
    SET-SIZE (mSendData) = 0.
END FINALLY.
```
This is similar to issue #1121 in DynamicObjectLeak rule

**Comment by Philippe-La** [comment]
> ### Case 1:
> Following code gives a false positive. Size of memptr is set inside the block and then released outside that block.
> 
> ```
> DEFINE VARIABLE mReceiveData AS MEMPTR   NO-UNDO.
> 
> IF TRUE THEN DO:
>     SET-SIZE(mReceiveData) = 100.
> END.
> 
> SET-SIZE(mReceiveData) = 0.
> ```
> 
> If release is done inside the same block or moved to finally-block, then issue is not raised.
> 
+1

**Comment by gquerret** [comment]
Fixed in develop branch, will be included in next release
